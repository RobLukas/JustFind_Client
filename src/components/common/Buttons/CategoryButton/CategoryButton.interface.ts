import { ReactChild } from 'react';

interface CategoryButtonProps {
  children: ReactChild;
}

export default CategoryButtonProps;
