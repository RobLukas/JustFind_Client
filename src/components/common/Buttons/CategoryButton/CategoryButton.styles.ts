import styled from 'styled-components';

const CategoryButton = styled.button`
  background: ${({ theme }) => theme.styles.secondary.background};
  color: ${({ theme }) => theme.styles.filter.button.color};
  min-width: 40px;
  cursor: pointer;
  margin-right: 15px;
  font-family: inherit;
  padding: 0 12px;
  box-shadow: none;
  text-transform: none;
  height: 35px;
  line-height: 35px;
  font-size: 12.6px;
  border-radius: 20px;
  text-align: center;
  display: flex;
  position: relative;
  text-decoration: none;
  letter-spacing: 0.5px;
  outline: 0;
  border: none;
  &:hover {
    background-color: rgba(255, 255, 255, 0.08);
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12),
      0 3px 1px -2px rgba(0, 0, 0, 0.2);
  }
`;

export default { CategoryButton };
