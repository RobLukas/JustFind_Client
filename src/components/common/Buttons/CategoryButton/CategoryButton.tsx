import React, { FC } from 'react';
import CategoryButtonProps from './CategoryButton.interface';
import S from './CategoryButton.styles';

const CategoryButton: FC<CategoryButtonProps> = ({
  children,
}: CategoryButtonProps) => {
  return <S.CategoryButton>{children}</S.CategoryButton>;
};

export default CategoryButton;
