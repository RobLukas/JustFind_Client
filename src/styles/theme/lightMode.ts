export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
