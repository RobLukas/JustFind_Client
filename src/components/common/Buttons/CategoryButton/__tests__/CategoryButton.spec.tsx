import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import lightMode from 'styles/theme/lightMode';
import CategoryButton from '../CategoryButton';

describe('<CategoryButton />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const children = 'text';

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <CategoryButton>{children}</CategoryButton>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
