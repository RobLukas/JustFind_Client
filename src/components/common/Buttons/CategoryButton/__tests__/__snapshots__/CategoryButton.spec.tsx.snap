// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryButton /> render correctly 1`] = `
.c0 {
  background: #fff;
  color: #777;
  min-width: 40px;
  cursor: pointer;
  margin-right: 15px;
  font-family: inherit;
  padding: 0 12px;
  box-shadow: none;
  text-transform: none;
  height: 35px;
  line-height: 35px;
  font-size: 12.6px;
  border-radius: 20px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  outline: 0;
  border: none;
}

.c0:hover {
  background-color: rgba(255,255,255,0.08);
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
}

<button
  class="c0"
>
  text
</button>
`;
